### DISCUSSION TITLE
qmenu, qinput and all in between

### DISCUSSION THREAD

**Discussion by AmosSzust** [quasarframework/quasar#17401] [General - Components / Directives / etc]
The below happens on both edge and chrome. I didn't try with Firefox.

I have a qmenu linked to a qinput
```
      <q-input
        rounded
        outlined
        v-model="searchText"
        autofocus
        @keyup.enter="doSearch"
        ><template v-slot:append>
          <q-icon
            name="search"
            size="lg"
            class="cursor-pointer"
            :color="searchText.length < 3 ? 'grey-5' : 'blue'"
            @click="doSearch"
            ><q-tooltip>search</q-tooltip></q-icon
          >
        </template>
        <q-menu fit auto-close>
          <q-list>
            <q-item
              v-for="user in usersList"
              v-bind:key="user.phoneNumber"
              @click="addUserToEvent(user)"
              clickable
              class="cursor-pointer"
            >
              <div class="row no-wrap">
                <q-icon :name="user.gender === 1 ? 'man' : 'woman'" size="md" />
                <div class="q-mt-xs">
                  {{ user.name }}: {{ user.phoneNumber }}
                </div>
              </div>
            </q-item>
          </q-list>
        </q-menu>
      </q-input>
```
When I run it locally, the obvious is happening, a request is made to the server, usersList is getting the response and the menu immediately appears. From now on, when I click inside the qinput, the menu disappears and re-appears as expected.

When running on my remote vps, the request is made, the response is back but the menu doesn't appear.
When I click inside the qinput, nothing happens but when I click again, the menu appears with the expected items. As if the first click hid it even though it was not shown. From now on, it behaves as expected, each click inside the qinput, shows and hides the menu.

I have no idea why there's a difference. I even tried setting a v-model to the qmenu and set it to true only after the response is back (using nextTick) but still, it behaves in the same problematic way on the vps (locally it works but it's always working).

I'm confused, please advice.

**Comment by ivanjaros** [comment]
you can fiddle with events in q-menu configuration but i'd just use v-model and keep it simple.

**Reply by AmosSzust** [reply]
Because it works locally and not remotely, I don't even know what to fiddle with.
v-model is the to go solution but even that didn't work (remotely, locally it worked)
The only thing I can think of is the difference in response time between working locally and remotely but that's just a guess that I have no way to prove.

**Reply by ivanjaros** [reply]
most likely it is "issue" of events. you have to configure the q-menu properly or it can look buggy, even though it is not. read the documentation, it's all there.

**Reply by ivanjaros** [reply]
as for not working in production, yes, it is most likely the delay in the data loading compared to localhost.
